### PR TITLE
CONTRAST-22497

### PR DESCRIPTION
**Pull request by serge17**
Now, the number of vulnerabilities found is logged to console before failing the build.

### CODE REVIEW

File: src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendHelper.java [hunk 1]
```diff
@@ -204,6 +206,62 @@ public static String buildAppVersionTag(Run<?, ?> build) {
         return build.getParent().getDisplayName() + "-" + build.getNumber();
     }
 
+    /**
```
Comment (gmoore) [inline src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendHelper.java:209]: Javadoc 👍

File: src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendHelper.java [hunk 2]
```diff
@@ -204,6 +206,62 @@ public static String buildAppVersionTag(Run<?, ?> build) {
...
+     */
+    public static String getVulnerabilityInfoString(Traces traces) {
+        StringBuilder info = new StringBuilder();
+        int note = 0;
+        int low = 0;
+        int medium = 0;
+        int high = 0;
+        int critical = 0;
+
+        if (traces != null && traces.getTraces() != null && !traces.getTraces().isEmpty()) {
```
Comment (gmoore) [inline src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendHelper.java]: This is a bit of a mess. "If traces is not null and traces get traces is not null and not traces is empty"

I get the checks here that you're being defensive and that's good. When things get complicated I like to spell it out explicitly with failing conditions.

```
if (traces == null || traces.getTraces() == null) {
  return info;
}

(go on with code)
```

File: src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendHelper.java [hunk 3]
```diff
@@ -204,6 +206,62 @@ public static String buildAppVersionTag(Run<?, ?> build) {
         return build.getParent().getDisplayName() + "-" + build.getNumber();
     }
 
+    /**
+     * Number of traces by severity
+     *
+     * @param traces
+     * @return String with the number of traces for each severity.
+     */
+    public static String getVulnerabilityInfoString(Traces traces) {
```
Comment (gmoore) [inline src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendHelper.java:215]: Need lots of tests here. Tests should specify:
* What gets returned if traces is null
* What gets returned if traces is empty
* What gets returned of there are several traces of different severities

**Review by gmoore — CHANGES_REQUESTED** [review]

File: src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendHelper.java (hunk 3 above)
Comment (serge17) [reply src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendHelper.java:215]: Tests added

**Review by gmoore — APPROVED** [review]

**Pull request merged**